Merge Oregon State Fair changes

5t0n3 opened the pull request:
Here's a short summary of what was changed:
* Homing and lowering of the new intake was implemented
* We replaced the PCM and some solenoid ports got swapped; these were updated in Constants.java
* A compressor toggle was added to the dashboard to avoid the battery voltage drops it causes, but ended up not being necessary
* Most `SmartDashboard.put*` calls were commented out to avoid cluttering the dashboard
* An RPM offset configurable during the match was added in case we were consistently over/undershooting

I'll probably do some clean-up before finalizing this as well.

nolanhergert committed:
Starting testing of climber, but disable intake

nolanhergert committed:
Update climber code, get roller running properly with belts

Commit:
Merge branch 'climber-test-disable-intake' into oregon-state-fair

Commit:
Make the intake actually stop after releasing right bumper (?)

Commit:
First wave of competition changes

New intake is now fully functional. Added a few commands to facilitate
the control of the intake.

Most SmartDashboard.put* calls were also commented out to avoid
cluttering the driver station.

Modified shooter feedforward a bit, but we still need to dial it in.

Commit:
Add a compressor toggle to the driver station

Commit:
More competition changes!

Added some delays to TaxiThenShoot auto, as well as added intake homing
as the first thing.

Adjusted climb timings.

Add an adjustable RPM offset to ShooterSubsystem.

Change the behavior for ShooterMath.calculateRPM so that if the
Limelight picks up a target outside of its calibrated range, it reverts
to the default RPM.

Commit:
Remove unused sections of RobotContainer

Commit:
Update climber ports after replacing the PCM

Commit:
RampThenShoot should now rumble both controllers

We had some trouble getting the secondary to rumble, so we should
probably check if that's an issue with the controller or just how we
rumble it.

The compressor is now enabled when telop starts (maybe) because we were
having some issues with it not running, but that might have been due to
it being at pressure.

Commit:
Change "winch" references to "lift" and delete unused commands

Intake and indexer constants were also split up in Constants.java.

Commit:
More general cleanup

A bunch of files (mostly commands) were deleted since they weren't used
anymore.

TurnToAngleProfiled was renamed to TurnDegrees to better reflect what it
actually does.

Smart Motion turn-related constants were removed as well since they were
unreliable & we got the NavX working pretty well.

I'm also trying to remove old TODO comments, so some of those were
deleted.

Commit:
Break up RobotContainer constructor into multiple methods

Commit:
Document ShooterMath

nolanhergert commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -29,10 +31,11 @@
-    public static final double kP = 1.2481e-11;
+    public static final double kP = 6e-5; // 1.7637e-8;
     public static final double kI = 0;
+    public static final double kIZone = 100; // rpm
```
This is most useful when kI is non-zero :)

5t0n3 replied:
Good point, I probably left that in when I was going to add an I which proved to be unnecessary :)

Commit:
Remove unused I zone & further document Constants.java

Commit:
Forward all necessary Limelight ports

nolanhergert commented on src/main/java/frc/robot/commands/utilities/TimeRotation.java:
```diff
@@ -10,6 +10,7 @@
   private double m_power;
   private final Timer m_timer = new Timer();
 
+  // TODO: Rewrite this command to be more accurate
```
What is inaccurate about it?

5t0n3 replied:
We don't account for the acceleration in the beginning which might throw off the calculated kV value? I'm not entire sure though.

nolanhergert replied:
We're talking about a simple TimeRotation command, which uses ArcadeDrive internally...it's not doing Profiled path following / feed forward.

nolanhergert reviewed (commented):
Looks good. I think you did the right thing of removing homing and incorporating it into the raising code. It's an implicit assumption that simplifies the code a little.

Commit:
Modified uncalibrated RPM behavior in ShooterMath

Rather than returning the default RPM for y-angles outside of the
calibrated range, it now returns the RPMs at the bound of that range (if
that makes sense).

Also applied Spotless, which reformatted a couple of unrelated files.

5t0n3 commented:
I'll probably also test this on the bot tomorrow to make sure I didn't break anything :)

Commit:
Add PDP/clearing of sticky faults on enabling test back in after updating firmware

5t0n3 commented:
Everything seems to be working (I didn't test climb but I didn't make any changes to it beyond during competition). Merging it now :)
